src/accel/SetupOptions.cc

```diff
@@ -118,10 +118,12 @@ void ProblemSetup::operator()(inp::Problem& p) const
         // Check if magnitude is zero
         auto field = u->get_field();
         auto field_val = norm(field.strength);
-        if (field_val > 0)
+	auto volumes = u->get_volumes();
+        if (field_val > 0 && !volumes.empty())
         {
             CELER_LOG(info) << "Using a uniform field: " << field_val << " [T]";
             p.field = std::move(field);
+	    p.volumes = std::move(volumes);
```

Comment:
It's OK to have a field without any volumes (an empty vector or no volumes means "all volumes").
```suggestion
        if (field_val > 0)
        {
            CELER_LOG(info) << "Using a uniform field: " << field_val << " [T]";
            p.field = std::move(field);
	    p.volumes = u->get_volumes();
```